Spara/bloggers oct 2019 (#1890)
pulumi/docs: content/blog/pulumi-bloggers-oct-2019/index.md

---
title: "Four Infrastructure as Code Blog Posts You Should Read"
authors: ["sophia-parafina"]
tags: ["Pulumi","bloggers",]
meta_image: "blog.png"
meta_desc: "Continuous delivery of Pulumi apps with Octopus Deploy "
date: "2019-10-24"
---

> We are always excited when people join the Infrastructure as Code community and write about their experiences using Pulumi. Here are four great posts ranging from an evaluation, a CI/CD tutorial, building a VM and Pulumi internals.

Reviewer: ```suggestion
We are always excited when people join the Infrastructure as Code community and write about their experiences. Here are four new posts that we think are great!
```

## HATech Compares Pulumi to Terraform

[Jon Hathway](https://twitter.com/hatechllc) of HATech published a white paper comparing Pulumi to Terraform and explained why HATech is evaluating Pulumi.

> Take for example the teams that are trying to collaborate and drive reliably deployed and maintained product innovations. Is it better to have languages that support each individual team’s needs, or is it better to have one language to support the entire workflow/pipeline? </BR>
> How about this thought experiment: Imagine your developers only spoke English, your QA team only spoke French, and the operations team only spoke Spanish? It would be carnage!</BR>
>You’d never hire that way. So why, when choosing DevOps tools, do people find it acceptable to have different languages, DSLs that only key individuals understand?”

Jon provides an in depth review of Pulumi and includes the [code samples](https://github.com/hatech/demo-pulumi-python) he used on GitHub. Download the white paper from [HATech](https://hatech.io/pulumi-v-terraform-whitepaper-lp/).

## VMWare VSphere and Pulumi

![peek definition](peek-definition.jpg)

> [Cody De Arkland](https://twitter.com/Codydearkland) demonstrates how to create a WMWare VSphere Virtual Machine by adding a ResourcePool, DataStore, NetworkStore and the template with Pulumi. He also sets the configuration values such as DNS settings, default gateway, domain, and hostname all in code. All of this runs on the ESXi virtualization server and vCenter (which is why the Datacenter ID is needed for configuration). His example is available on [GitHub}(https://github.com/codyde/pulumi-vsphere-ts). We like Cody’s caveat that his post is not an endorsement but call to explore new tools. In his conclusion, he nails home the point why Infrastructure as Code is the future of DevOps.

Reviewer: ```suggestion
[Cody De Arkland](https://twitter.com/Codydearkland) demonstrates how to create a VMWare VSphere Virtual Machine by adding a ResourcePool, DataStore, and NetworkStore with Pulumi. Cody also sets configuration values such as DNS settings, default gateway, domain, and hostname all in code! All of this runs on the ESXi virtualization server and vCenter (which is why the Datacenter ID is needed for configuration). His example is available on [GitHub](https://github.com/codyde/pulumi-vsphere-ts). We like Cody’s caveat that his post is not an endorsement but call to explore new tools. In his conclusion, he nails home the point why Infrastructure as Code is the future of DevOps.
```

---
title: "Four Infrastructure as Code Blog Posts You Should Read"
authors: ["sophia-parafina"]
tags: ["Pulumi","bloggers",]
meta_image: "blog.png"
meta_desc: "Continuous delivery of Pulumi apps with Octopus Deploy "
date: "2019-10-24"
---

> We are always excited when people join the Infrastructure as Code community and write about their experiences. Here are four new posts that we think are great!

Reviewer 2: Is there any way to edit the preview? The preview seems to go into the first section which is a tad strange.

![image](https://user-images.githubusercontent.com/8304499/67536790-a163ba80-f68d-11e9-92e8-59b9e80d3052.png)

Author: Unsure why the preview looks like that.

## CircleCI on Google Kubernetes Engine (GKE)

> [Angel Rivera](https://twitter.com/punkdata) shows how to set up CircleCI on GKE with a [step-by-step tutorial](https://circleci.com/blog/automate-releases-from-pipelines-using-infrastructure-as-code/). Angel demonstrates how to integrate Infrastructure as Code solutions, such as Pulumi, into CI/CD pipelines. The tutorial builds a container with a Pulumi project with a sample app written in Python that deploys Kubernetes. The tutorial shows how you can incorporate your infrastructure into your CI/CD pipeline. The code for the tutorial is available in [GitHub](https://github.com/datapunkz/orb-pulumi-gcp).

Reviewer 2: `integrate Infrastructure as Code solutions, such as Pulumi, into CI/CD pipelines`
and 
`The tutorial shows how you can incorporate your infrastructure into your CI/CD pipeline.`
are a bit repetitive!

Reviewer 2: should it be "on GitHub" instead of "in"? (Could totally be wrong here)

## HATech Compares Pulumi to Terraform

> [Jon Hathway](https://twitter.com/hatechllc) of HATech published a white paper comparing Pulumi to Terraform and explained why HATech is evaluating Pulumi.

Reviewer 2: Should this be `explaining why HATech is evaluating Pulumi.` to mimic the `comparing`?

Author: the quote is the explanation, but good catch. I changed it to be more explicit.

> > Take for example the teams that are trying to collaborate and drive reliably deployed and maintained product innovations. Is it better to have languages that support each individual team’s needs, or is it better to have one language to support the entire workflow/pipeline?

Reviewer 2: * Are these three separate quotes? On the site they sort of blend together. 
* I also think an intro to these would be great. e.g. "Here are some snippets from the whitepaper:" or "Here are some of my favorite parts:"

>
> How about this thought experiment: Imagine your developers only spoke English, your QA team only spoke French, and the operations team only spoke Spanish? It would be carnage!
>
>You’d never hire that way. So why, when choosing DevOps tools, do people find it acceptable to have different languages, DSLs that only key individuals understand?”

Jon provides an in-depth review of Pulumi and includes the [code samples](https://github.com/hatech/demo-pulumi-python) he used on GitHub. Download the white paper from [HATech](https://hatech.io/pulumi-v-terraform-whitepaper-lp/).

> ## VMWare VSphere and Pulumi

Reviewer 2: Is there a link to this post?

Author: There's a download link for the paper

Reviewer 2: Theres a link for downloading the VMWare post?

![peek definition](peek-definition.jpg)

> [Cody De Arkland](https://twitter.com/Codydearkland) demonstrates how to create a VMWare VSphere Virtual Machine by adding a ResourcePool, DataStore, NetworkStore and the template with Pulumi. He also sets the configuration values such as DNS settings, default gateway, domain, and hostname all in code. All of this runs on the ESXi virtualization server and vCenter (which is why the Datacenter ID is needed for configuration). His example is available on [GitHub}(https://github.com/codyde/pulumi-vsphere-ts). We like Cody’s caveat that his post is not an endorsement but call to explore new tools. In his conclusion, he nails home the point why Infrastructure as Code is the future of DevOps.

Reviewer 2: * I could be missing some context, but what is the template referred to here? `and the template with Pulumi.`
* The link is not rendering properly due to brackets `[GitHub}(https://github.com/codyde/pulumi-vsphere-ts)`
* `We like Cody’s caveat that his post is not an endorsement but call to explore new tools.` is a bit of a strange add in.. although I didnt have a link to the actual article so maybe its not strange?

Author: I added 'virtual machine' template
fixed link
Cody says the blog wasn't an endorsement but something to add to your tool box.

## Pulumi Architecture and Examples

For the francophones in our community, Thomas Schersach has written a three-part series that includes deploying Kubernetes and a serverless application. The [first post](https://blog.octo.com/decouvrir-les-cloud-native-languages-avec-pulumi/) is an overview of how Pulumi works as well as the role of Providers for resource deployments. The [second article](https://blog.octo.com/pulumi-par-la-pratique-kubernetes/) demonstrates how to deploy Amazon’s Elastic Kubernetes Services with Pulumi. In the [third installment](https://blog.octo.com/pulumi-par-la-pratique-serverless/), Thomas shows how to deploy a Lambda serverless function. Example code is available on [GitHub](https://github.com/Tirke/try-pulumi). Overall a great set of articles to get you up and running on Pulumi; and if you don’t speak or read French, Google Translate is your [friend](https://translate.google.com/translate?sl=fr&tl=en&u=https://blog.octo.com/author/thomas-schersach-thom/).

## Bloggers Welcome

> If you write about your experience using Pulumi, let us know by tagging us on [Twitter](pulumicorp).

Reviewer 2: This twitter link doesnt appear to work.

Author: link fixed
